Vorkath Counter plugin

goosoodude opened the pull request:
Simple Vorkath attack counter, didn't want to go too over the top for this one.

Gamer1120 commented on runelite-api/src/main/java/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Main-Class: net.runelite.client.RuneLite
```
I'd suggest you delete this file.

Gamer1120 commented on runelite-client/src/main/java/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
```
I would also suggest to delete this file.

deathbeam replied:
These manifest files are still not deleted

Gamer1120 commented on runelite-client/src/main/java/net/runelite/client/plugins/vorkath/VorkathAttack.java:
```diff
@@ -0,0 +1,24 @@
+package net.runelite.client.plugins.vorkath;
```
Missing copyright header

Gamer1120 commented on runelite-client/src/main/java/net/runelite/client/plugins/vorkath/VorkathOverlay.java:
```diff
@@ -0,0 +1,61 @@
+package net.runelite.client.plugins.vorkath;
```
Missing another copyright header

Gamer1120 commented on runelite-client/src/main/java/net/runelite/client/plugins/vorkath/VorkathPlugin.java:
```diff
@@ -0,0 +1,106 @@
+package net.runelite.client.plugins.vorkath;
```
Another missing copyright notice

Gamer1120 commented on runelite-client/src/main/java/net/runelite/client/plugins/vorkath/VorkathPlugin.java:
```diff
@@ -0,0 +1,106 @@
+			return;
+		}
+
+		NPC vorkath = findVorkath();
```
Maybe it's worth checking whether the player is in the Vorkath instance before checking all NPC's?

Infinitay commented:
My only suggestion would be to stray away from the Scheduler and instead use the AnimationChanged event

goosoodude commented:
I've added most of the things suggested and fixed an issue I found while using the plugin, however I'm debating switching to the AnimationChanged event. I know the Scheduler isn't the greatest thing to be using, however the plugin is working quite well as-is. If there are any issues someone finds, I could update it, but I don't see any arising.

Infinitay commented:
Glad there isn't any problems, in case you do end up using the event, as well as it's benefit, you don't have to query for the NPC, you could check the Actors name from `AnimationChanged::getActor`.

Oh one last thing, `private VorkathAttack attack;`, with lombok, you could add the annotation `@Getter` so you don't have to create a get method for the variable, simplifies things :p Then again, IntelliJ already makes it so easy for get/set methods

goosoodude commented:
I've completed all requested changes. I found an additional issue using the scheduler, and updated the plugin to use the AnimationChanged event instead. I've been testing the updated plugin for about 2 hours now and everything is working now, all attacks are counted. Unless there's anything else that needs to be fixed, I'm pretty much done working on this plugin at this point.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/vorkath/VorkathOverlay.java:
```diff
@@ -0,0 +1,86 @@
+	public Dimension render(Graphics2D graphics)
+	{
+		panelComponent.getChildren().clear();
+		panelComponent.getChildren().add(TitleComponent.builder()
```
Use some images instead of this text, in style of cerb, jad plugin

goosoodude commented:
Closed because I'm quite busy with exams and can't complete the requested changes at this time.
